src/charm.py

```diff
@@ -300,20 +340,19 @@ def _on_update_status(self, _) -> None:
                 self.peers.data[unit].get("member-state", "unknown") for unit in self.peers.units
             }
 
-            if all_states == {"offline"} and self.unit.is_leader():
-                # All instance are off, reboot cluster from outage from the leader unit
-
+            if all_states in [{"offline"}, set()] and self.unit.is_leader():
```

Comment:
@carlcsaposs-canonical  discovered this in the k8s operator, and initialized `all_states = set(['offline'])` as we've already established that the current unit is offline. i like that better as it is more readable